### PR TITLE
Add deposit switch product

### PR DESCRIPTION
**Pull request by jeffzwang**

### CODE REVIEW

File: android/src/main/java/com/plaid/PlaidModule.kt [hunk 1]
```diff
@@ -89,7 +90,7 @@ class PlaidModule internal constructor(reactContext: ReactApplicationContext) :
       return null
     }
 
-    if (!token.startsWith(LINK_TOKEN_PREFIX)) {
+    if (!token.startsWith(LINK_TOKEN_PREFIX) && !token.startsWith(LINK_TOKEN_PREFIX)) {
```
Comment (zsweigart) [inline android/src/main/java/com/plaid/PlaidModule.kt:93]:
```suggestion
    if (!token.startsWith(LINK_TOKEN_PREFIX) && !token.startsWith(DEPOSIT_SWITCH_TOKEN_PREFIX)) {
```

**Review by zsweigart — CHANGES_REQUESTED** [review]
We're updating types soon, so this will have to be updated.

**Review by zsweigart — APPROVED** [review]

**Review by amytang0 — CHANGES_REQUESTED** [review]
RC for requiring the underlying SDKs

**Comment by JeroenMols** [comment]
Closing this PR as we are in the midst of a typescript conversion.

- The non typescript SDK 6.0.4 already has `deposit_switch` added
- The typescript SDK 7.x will get `deposit_switch` through https://github.com/plaid/react-native-plaid-link-sdk/pull/269

**Pull request closed without merging**